src/utils/patches.js

```diff
@@ -387,3 +392,50 @@ export const addPrefixToPatch = (prefix, patch) => ({
   ...patch,
   path: `${prefix}${patch.path}`,
 });
+
+const getWrappedDeviceType = device => {
+  let type;
+  if (isNic(device)) {
+    type = DEVICE_TYPE_INTERFACE;
+  } else if (isDisk(device)) {
+    type = DEVICE_TYPE_DISK;
+  }
+
+  return type;
+};
+
+const getDeviceIndex = (vm, deviceName, deviceType) => {
+  const devices = deviceType === DEVICE_TYPE_INTERFACE ? getInterfaces(vm) : getDisks(vm);
+  return findIndex(devices, d => d.name === deviceName);
+};
+
+export const getDeviceBootOrderPatch = (vm, removedDeviceType, removedDeviceName) => {
+  const patches = [];
+  const removedDevice = get(getDevices(vm), removedDeviceType, []).find(dev => dev.name === removedDeviceName);
+  const removedDeviceBootOrder = get(removedDevice, 'bootOrder', -1);
+  const removedDeviceIndex = getDeviceIndex(vm, removedDeviceName, removedDeviceType);
+  const sortedBootableDevices = getBootableDevicesInOrder(vm);
+
+  sortedBootableDevices.forEach(device => {
+    const bootOrder = get(device, 'value.bootOrder', -1);
+
+    if (bootOrder !== -1 && bootOrder > removedDeviceBootOrder) {
+      const deviceType = getWrappedDeviceType(device);
+      const currDevIndex = getDeviceIndex(vm, device.value.name, deviceType);
+      // Account for index change once device is removed
+      const deviceIndex =
+        deviceType === removedDeviceType && currDevIndex > removedDeviceIndex ? currDevIndex - 1 : currDevIndex;
+
+      if (deviceIndex !== -1 && deviceType) {
+        const patch = {
+          op: 'replace',
+          path: `/spec/template/spec/domain/devices/${deviceType}/${deviceIndex}/bootOrder`,
```

Comment:
Oh! I see now, why is that needed. I think it would be probably better to decouple the type from the path fragment. I would rather use the singular for type and then do the mapping of the type to the path fragment. Quick and readable way is to use object for mappings for example.

this way, the functionality will not break when somebody decides to refactor the type

Reply:
Agreed and done! Could you please check the way I implemented your suggestion to make sure this is what you were thinking?

Reply:
yup, looks good! 

Although I would rather name it DEVICE_TYPE_INTERFACE or something like that. INTERFACE/DISK seems a bit generic.

Reply:
Agreed and done! =)